### PR TITLE
Added logsumexp to backend.

### PR DESCRIPTION
**Pull request by phipleg**
Adds the method logsumexp(x, axis, keepdims) to the backend.

Tensorflow (resp. Theano) offers an optimization for computing `log(sum(exp(x)))` as `x_max + log(sum(exp(x - x_max)))` that improves numerical stability for large values. It is exposed by `tf.reduce_logsumexp` (resp. internal graph optimization).

Having this functionality simplifies building a Conditional Random Field with Keras (see PR #4621).

### CODE REVIEW

File: keras/backend/tensorflow_backend.py [hunk 1]
```diff
@@ -1274,6 +1274,13 @@ def log(x):
     return tf.log(x)
 
 
+def logsumexp(x, axis=None, keepdims=False):
+    '''Returns `log(sum(exp(x), axis=axis, keepdims=keepdims))` with improved numerical stability.
```
Comment (fchollet) [inline keras/backend/tensorflow_backend.py]: The one-line docstring summary should fit within 80 chars.

Both added functions need a compliant docstring (`# Arguments`, `# Returns`).

Comment (phipleg) [inline keras/backend/tensorflow_backend.py]: Thanks. It is fixed now.

**Comment by fchollet** [comment]
LGTM
